tests/test_init_and_backend.py

```diff
@@ -0,0 +1,139 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import filecmp
+import tempfile
+from pathlib import Path
+
+import mkosi
+import mkosi.backend as backend
+import pytest
+
+
+def test_workspace():
+    assert backend.workspace(Path("/home/folder/mkosi/mkosi")) == Path("/home/folder/mkosi")
+    assert backend.workspace(Path("/home/../home/folder/mkosi/mkosi")) == Path("/home/../home/folder/mkosi")
+    assert backend.workspace(Path("/")) == Path("/")
+    assert backend.workspace(Path()) == Path()
+
+def test_fedora_release_cmp():
+    assert mkosi.fedora_release_cmp("rawhide", "rawhide") == 0
+    assert mkosi.fedora_release_cmp("32", "32") == 0
+    assert mkosi.fedora_release_cmp("33", "32") > 0
+    assert mkosi.fedora_release_cmp("30", "31") < 0
+    assert mkosi.fedora_release_cmp("-1", "-2") > 0
+    assert mkosi.fedora_release_cmp("1", "-2") > 0
+    with pytest.raises(ValueError):
+        mkosi.fedora_release_cmp("literal", "rawhide")
+
+def test_strip_suffixes():
+    assert mkosi.strip_suffixes(Path("home/test.zstd")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.xz")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.raw")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.tar")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.cpio")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home/test.qcow2")) == Path("home/test")
+    assert mkosi.strip_suffixes(Path("home.xz/test.xz")) == Path("home.xz/test")
+    assert mkosi.strip_suffixes(Path("home.xz/test")) == Path("home.xz/test")
+    assert mkosi.strip_suffixes(Path("home.xz/test.txt")) == Path("home.xz/test.txt")
+
+def test_copy_file():
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        dir_path = Path(tmp_dir)
+        file_1 = Path(dir_path) / "file_1.txt"
+        file_2 = Path(dir_path) / "file_2.txt"
+        file_1.touch()
+        file_2.touch()
+
+        # Copying two empty files.
+        mkosi.copy_file(file_1, file_2)
+        assert (filecmp.cmp(file_1, file_2)) == True
```

Comment:
Asserts like these are a tautology: `assert (filecmp.cmp(file_1, file_2)) == True`

You can write this as `assert filecmp.cmp(file_1, file_2)`